### PR TITLE
Run clippy 

### PR DESCRIPTION
**Pull request by hellow554**
I ran clippy on your project and solved most of the issues. 
Also I ran rustfmt (you should do this with a git hook IMHO, so it is always formatted).

I know this is a huge PR and I understand if you are not willing to accept all changes. Please say what you like or dislike and I will change that.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -253,7 +253,7 @@ dependencies = [
  "i3ipc 0.8.2 (registry+https://github.com/rust-lang/crates.io-index)",
  "inotify 0.5.1 (registry+https://github.com/rust-lang/crates.io-index)",
  "lazy_static 1.0.0 (registry+https://github.com/rust-lang/crates.io-index)",
- "libpulse-binding 2.2.1 (registry+https://github.com/rust-lang/crates.io-index)",
+ "libpulse-binding 2.2.6 (registry+https://github.com/rust-lang/crates.io-index)",
```
Comment (atheriel) [inline Cargo.lock]: Did you intend to update this dependency?

**Comment by atheriel** [comment]
Running `rustfmt` on the codebase has come up before, and has some drawbacks. But I'm happy to accept the changes you made due to clippy.

**Comment by hellow554** [comment]
pushed and rebased :)

**Comment by atheriel** [comment]
Thanks!

**Pull request merged**